logr.go

```diff
@@ -246,6 +247,26 @@ func (lgr *Logr) SetMetricsCollector(collector MetricsCollector, updateFreqMilli
 // this function either blocks or the log record is dropped, depending on
 // the result of calling `OnQueueFull`.
 func (lgr *Logr) enqueue(rec *LogRec) {
+	// we also limit the message
+	rec.msg = LimitString(rec.msg, lgr.options.maxFieldLen)
```

Comment:
Can we default the limit to -1 (unlimited) and skip all this unless a limit has been set?  I don't want to impact performance here unless someone explicitly wants limits.